volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+            return unpack('<B', mft_record[offset:offset+1])[0]
+        elif data_type == 'int':
+            return unpack('<I', mft_record[offset:offset+4])[0]
+
+    @classmethod
+    def human_date(self, datetime_object: int) -> str:
+        """Converts a windows epoch to a date time string with a fixed format
+        
+        Args:
+            datetime_object: windows epoch time
+
+        Returns:
+            str: strftime of the windows epoch in UTC
+
+        """
+        dtg = conversion.wintime_to_datetime(datetime_object)
+        return dtg.strftime('%Y-%m-%d %H:%M:%S %z')
+
+    @classmethod
+    def parse_mft_record(self, mft_record: bytes) -> Dict:
+        """Takes an MFT Record and attempts to parse, MFT, SI and FN attributes
+        
+        Args:
+            mft_record: 1024 bytes starting from header value as returned by layer read
+
+        Returns:
+            Dict: a Dictionary that contains the Parse MFT Record
+        """
+        # https://github.com/Invoke-IR/ForensicPosters
+
+        flags = self.unpack_data(mft_record, 22, 'unsigned short')
+        file_type = MFT_FLAGS.get(flags, 'Unknown')
+
+        mft_entry = {
+            "signature": mft_record[:4].decode(),
+            "FixupArrayOffset": self.unpack_data(mft_record, 4, 'unsigned short'),
+            "NumFixupEntries": self.unpack_data(mft_record, 6, 'unsigned short'),
+            "LSN": self.unpack_data(mft_record, 8,  'unsigned long long'),
+            "SequenceValue": self.unpack_data(mft_record, 16, 'unsigned short'),
+            "link_count": self.unpack_data(mft_record, 18, 'unsigned short'),
+            "FirstAttrOffset": self.unpack_data(mft_record, 20, 'unsigned short'),
+            "flags": file_type,
+            "record_number": self.unpack_data(mft_record, 44, 'unsigned long'),
+            "attributes": {
+            "SI": {},
+            "FN": []
+        }
+        }
+
+        attr_offset = mft_entry['FirstAttrOffset']
+        # Check at most for 6 entries
+        for i in range(6):
+            # If we attempt to overread the entry continue out
+            if attr_offset > 1000:
+                continue
+
+            # attr_header
+            attr_type = self.unpack_data(mft_record, attr_offset, 'int')
+            attr_len = self.unpack_data(mft_record, attr_offset+4, 'int')
+
+            # As we look for strucutres of header + 1K we can not unpack non resident structures
+            nr_flag = self.unpack_data(mft_record, attr_offset+8, 'unsigned char')
+
+            # Skip headers
+            attr_data = attr_offset+24 # Len of Common and Resident Headers
+
+            if attr_type in ATTRIBUTE_TYPE_ID:
+                vollog.debug(f'Found Attribute {ATTRIBUTE_TYPE_ID[attr_type]}')
+                
+                if ATTRIBUTE_TYPE_ID[attr_type] == 'STANDARD_INFORMATION':
+                    creation_time_win = self.unpack_data(mft_record, attr_data,  'unsigned long long')
```

Comment:
This also feels like it would be better as a reusable type that returns an object, rather than manually populated a structure in the code.